The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/14534933096/job/40781448268) [(plain)](https://github.com/rust-lang/rust/commit/3abfd82625b982e631e022e5bd8ca7e389388a84/checks/40781448268/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
tests/fail/function_calls/exported_symbol_bad_unwind2.rs (revision `definition`) ... ok
tests/fail/function_calls/exported_symbol_bad_unwind2.rs (revision `both`) ... ok

FAILED TEST: tests/fail/panic/tls_macro_const_drop_panic.rs
command: MIRI_ENV_VAR_TEST="0" MIRI_TEMP="/tmp/miri-uitest-uVQ2ij" RUST_BACKTRACE="1" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/miri" "--error-format=json" "--sysroot=/checkout/obj/build/x86_64-unknown-linux-gnu/miri-sysroot" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zui-testing" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/miri_ui/0/tests/fail/panic" "tests/fail/panic/tls_macro_const_drop_panic.rs" "-Zmiri-backtrace=full" "--edition" "2021"

error: actual output differed from expected
Execute `./miri test --bless` to update `tests/fail/panic/tls_macro_const_drop_panic.stderr` to the actual output
--- tests/fail/panic/tls_macro_const_drop_panic.stderr
+++ <stderr output>
---

thread 'main' panicked at tests/fail/panic/tls_macro_const_drop_panic.rs:12:9:
ow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
note: in Miri, you may have to set `MIRIFLAGS=-Zmiri-env-forward=RUST_BACKTRACE` for the environment variable to have an effect
fatal runtime error: thread local panicked on drop, aborting
error: abnormal termination: the program aborted execution
##[error]  --> /checkout/library/std/src/sys/pal/unix/mod.rs:367:14
   |
LL |     unsafe { libc::abort() }
   |              ^^^^^^^^^^^^^ the program aborted execution
   |
   = note: BACKTRACE:
   = note: inside `std::sys::pal::unix::abort_internal` at /checkout/library/std/src/sys/pal/unix/mod.rs:367:14: 367:27
   = note: inside `<std::sys::thread_local::abort_on_dtor_unwind::DtorUnwindGuard as std::ops::Drop>::drop` at /checkout/library/std/src/rt.rs:50:13: 50:41
   = note: inside `std::ptr::drop_in_place::<std::sys::thread_local::abort_on_dtor_unwind::DtorUnwindGuard> - shim(Some(std::sys::thread_local::abort_on_dtor_unwind::DtorUnwindGuard))` at /checkout/library/core/src/ptr/mod.rs:524:1: 524:56
   = note: inside `std::sys::thread_local::abort_on_dtor_unwind::<{closure@std::sys::thread_local::native::eager::destroy<NoisyDrop>::{closure#0}}>` at /checkout/library/std/src/sys/thread_local/mod.rs:200:1: 200:2
   = note: inside `std::sys::thread_local::native::eager::destroy::<NoisyDrop>` at /checkout/library/std/src/sys/thread_local/native/eager.rs:64:5: 72:7
   = note: inside `std::sys::thread_local::destructors::list::run` at /checkout/library/std/src/sys/thread_local/destructors/list.rs:34:21: 34:28
   = note: inside `std::sys::thread_local::guard::key::enable::run` at /checkout/library/std/src/sys/thread_local/guard/key.rs:22:13: 22:31

error: aborting due to 1 previous error



full stderr:

thread 'main' panicked at tests/fail/panic/tls_macro_const_drop_panic.rs:12:9:
ow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
note: in Miri, you may have to set `MIRIFLAGS=-Zmiri-env-forward=RUST_BACKTRACE` for the environment variable to have an effect
fatal runtime error: thread local panicked on drop, aborting
error: abnormal termination: the program aborted execution
##[error]  --> /checkout/library/std/src/sys/pal/unix/mod.rs:367:14
   |
LL |     unsafe { libc::abort() }
   |              ^^^^^^^^^^^^^ the program aborted execution
   |
   = note: BACKTRACE:
   = note: inside `std::sys::pal::unix::abort_internal` at /checkout/library/std/src/sys/pal/unix/mod.rs:367:14: 367:27
   = note: inside `<std::sys::thread_local::abort_on_dtor_unwind::DtorUnwindGuard as std::ops::Drop>::drop` at /checkout/library/std/src/rt.rs:50:13: 50:41
   = note: inside `std::ptr::drop_in_place::<std::sys::thread_local::abort_on_dtor_unwind::DtorUnwindGuard> - shim(Some(std::sys::thread_local::abort_on_dtor_unwind::DtorUnwindGuard))` at /checkout/library/core/src/ptr/mod.rs:524:1: 524:56
   = note: inside `std::sys::thread_local::abort_on_dtor_unwind::<{closure@std::sys::thread_local::native::eager::destroy<NoisyDrop>::{closure#0}}>` at /checkout/library/std/src/sys/thread_local/mod.rs:200:1: 200:2
   = note: inside `std::sys::thread_local::native::eager::destroy::<NoisyDrop>` at /checkout/library/std/src/sys/thread_local/native/eager.rs:64:5: 72:7
   = note: inside `std::sys::thread_local::destructors::list::run` at /checkout/library/std/src/sys/thread_local/destructors/list.rs:34:21: 34:28
   = note: inside `std::sys::thread_local::guard::key::enable::run` at /checkout/library/std/src/sys/thread_local/guard/key.rs:22:13: 22:31

error: aborting due to 1 previous error


full stdout:



FAILED TEST: tests/fail/panic/tls_macro_drop_panic.rs
command: MIRI_ENV_VAR_TEST="0" MIRI_TEMP="/tmp/miri-uitest-uVQ2ij" RUST_BACKTRACE="1" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/miri" "--error-format=json" "--sysroot=/checkout/obj/build/x86_64-unknown-linux-gnu/miri-sysroot" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zui-testing" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/miri_ui/0/tests/fail/panic" "tests/fail/panic/tls_macro_drop_panic.rs" "--edition" "2021"

error: actual output differed from expected
Execute `./miri test --bless` to update `tests/fail/panic/tls_macro_drop_panic.stderr` to the actual output
--- tests/fail/panic/tls_macro_drop_panic.stderr
+++ <stderr output>
---

thread 'main' panicked at tests/fail/panic/tls_macro_drop_panic.rs:10:9:
ow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
note: in Miri, you may have to set `MIRIFLAGS=-Zmiri-env-forward=RUST_BACKTRACE` for the environment variable to have an effect
fatal runtime error: thread local panicked on drop, aborting
error: abnormal termination: the program aborted execution
##[error]  --> /checkout/library/std/src/sys/pal/unix/mod.rs:367:14
   |
LL |     unsafe { libc::abort() }
   |              ^^^^^^^^^^^^^ the program aborted execution
   |
   = note: BACKTRACE:
   = note: inside `std::sys::pal::unix::abort_internal` at /checkout/library/std/src/sys/pal/unix/mod.rs:367:14: 367:27
   = note: inside `<std::sys::thread_local::abort_on_dtor_unwind::DtorUnwindGuard as std::ops::Drop>::drop` at /checkout/library/std/src/rt.rs:50:13: 50:41
   = note: inside `std::ptr::drop_in_place::<std::sys::thread_local::abort_on_dtor_unwind::DtorUnwindGuard> - shim(Some(std::sys::thread_local::abort_on_dtor_unwind::DtorUnwindGuard))` at /checkout/library/core/src/ptr/mod.rs:524:1: 524:56
   = note: inside `std::sys::thread_local::abort_on_dtor_unwind::<{closure@std::sys::thread_local::native::lazy::destroy<NoisyDrop>::{closure#0}}>` at /checkout/library/std/src/sys/thread_local/mod.rs:200:1: 200:2
   = note: inside `std::sys::thread_local::native::lazy::destroy::<NoisyDrop>` at /checkout/library/std/src/sys/thread_local/native/lazy.rs:93:5: 99:7
   = note: inside `std::sys::thread_local::destructors::list::run` at /checkout/library/std/src/sys/thread_local/destructors/list.rs:34:21: 34:28
   = note: inside `std::sys::thread_local::guard::key::enable::run` at /checkout/library/std/src/sys/thread_local/guard/key.rs:22:13: 22:31

error: aborting due to 1 previous error



full stderr:

thread 'main' panicked at tests/fail/panic/tls_macro_drop_panic.rs:10:9:
ow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
note: in Miri, you may have to set `MIRIFLAGS=-Zmiri-env-forward=RUST_BACKTRACE` for the environment variable to have an effect
fatal runtime error: thread local panicked on drop, aborting
error: abnormal termination: the program aborted execution
##[error]  --> /checkout/library/std/src/sys/pal/unix/mod.rs:367:14
   |
LL |     unsafe { libc::abort() }
   |              ^^^^^^^^^^^^^ the program aborted execution
   |
   = note: BACKTRACE:
   = note: inside `std::sys::pal::unix::abort_internal` at /checkout/library/std/src/sys/pal/unix/mod.rs:367:14: 367:27
   = note: inside `<std::sys::thread_local::abort_on_dtor_unwind::DtorUnwindGuard as std::ops::Drop>::drop` at /checkout/library/std/src/rt.rs:50:13: 50:41
   = note: inside `std::ptr::drop_in_place::<std::sys::thread_local::abort_on_dtor_unwind::DtorUnwindGuard> - shim(Some(std::sys::thread_local::abort_on_dtor_unwind::DtorUnwindGuard))` at /checkout/library/core/src/ptr/mod.rs:524:1: 524:56
   = note: inside `std::sys::thread_local::abort_on_dtor_unwind::<{closure@std::sys::thread_local::native::lazy::destroy<NoisyDrop>::{closure#0}}>` at /checkout/library/std/src/sys/thread_local/mod.rs:200:1: 200:2
   = note: inside `std::sys::thread_local::native::lazy::destroy::<NoisyDrop>` at /checkout/library/std/src/sys/thread_local/native/lazy.rs:93:5: 99:7
   = note: inside `std::sys::thread_local::destructors::list::run` at /checkout/library/std/src/sys/thread_local/destructors/list.rs:34:21: 34:28
   = note: inside `std::sys::thread_local::guard::key::enable::run` at /checkout/library/std/src/sys/thread_local/guard/key.rs:22:13: 22:31

error: aborting due to 1 previous error


full stdout:
---

Location:
   /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ui_test-0.29.2/src/lib.rs:369

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
error: test failed, to rerun pass `--test ui`

Caused by:
  process didn't exit successfully: `/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/ui-2335cdebe069a85a` (exit status: 1)
Command has failed. Rerun with -v to see more details.
Build completed unsuccessfully in 0:04:39
  local time: Fri Apr 18 12:44:32 UTC 2025
  network time: Fri, 18 Apr 2025 12:44:33 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/std/src/rt.rs
+++ b/library/std/src/rt.rs
@@ -46,7 +46,7 @@ macro_rules! rtprintpanic {
 macro_rules! rtabort {
     ($($t:tt)*) => {
         {
-            rtprintpanic!("fatal runtime error: {}\n", format_args!($($t)*));
+            rtprintpanic!("fatal runtime error: {}, aborting\n", format_args!($($t)*));
             crate::sys::abort_internal();
         }
     }

--- a/src/tools/miri/tests/fail/panic/tls_macro_const_drop_panic.stderr
+++ b/src/tools/miri/tests/fail/panic/tls_macro_const_drop_panic.stderr
@@ -1,7 +1,7 @@
 
 thread $NAME panicked at tests/fail/panic/tls_macro_const_drop_panic.rs:LL:CC:
 ow
-fatal runtime error: thread local panicked on drop
+fatal runtime error: thread local panicked on drop, aborting
 error: abnormal termination: the program aborted execution
 
 error: aborting due to 1 previous error

--- a/src/tools/miri/tests/fail/panic/tls_macro_drop_panic.stderr
+++ b/src/tools/miri/tests/fail/panic/tls_macro_drop_panic.stderr
@@ -1,7 +1,7 @@
 
 thread $NAME panicked at tests/fail/panic/tls_macro_drop_panic.rs:LL:CC:
 ow
-fatal runtime error: thread local panicked on drop
+fatal runtime error: thread local panicked on drop, aborting
 error: abnormal termination: the program aborted execution
 
 error: aborting due to 1 previous error

--- a/tests/ui/runtime/rt-explody-panic-payloads.rs
+++ b/tests/ui/runtime/rt-explody-panic-payloads.rs
@@ -27,6 +27,6 @@ fn main() {
             // by QEMU in the stderr whenever a core dump happens. Remove it before the check.
             v.strip_suffix("qemu: uncaught target signal 6 (Aborted) - core dumped\n").unwrap_or(v)
         })
-        .map(|v| { v.ends_with("fatal runtime error: drop of the panic payload panicked\n") })
+        .map(|v| v.ends_with("fatal runtime error: drop of the panic payload panicked, aborting\n"))
         .unwrap_or(false));
 }

